### ISSUE TITLE
bootcfg sometimes cannot parse ignition yaml files

### ISSUE THREAD

**Issue by urzds** [coreos/coreos-baremetal#86]
My `ignition/coreos-install.yaml` file:
```yaml
---
ignition_version: 1
systemd:
  units:
    - name: coreos-install.service
      enable: true
      contents: |
        [Unit]
        Require=network-online.target
        After=network-online.target
        [Service]
        Type=oneshot
        ExecStart=/usr/bin/curl {{.ignition_endpoint}}?{{.query}}&os=installed -o ignition.json
        ExecStart=/usr/bin/coreos-install -d /dev/sda -C {{.coreos_channel}} -V {{.coreos_version}} -i ignition.json
        ExecStart=/usr/bin/docker run -ti --device=/dev/ipmi0 urzds/ipmitool chassis bootdev disk options=persistent
        ExecStart=/usr/bin/udevadm settle
        ExecStart=/usr/bin/systemctl reboot
        [Install]
        WantedBy=multi-user.target
{{ if .ssh_authorized_keys }}
passwd:
  users:
    - name: core
      ssh_authorized_keys:
        {{ range $element := .ssh_authorized_keys }}
        - {{$element}}
        {{end}}
{{end}}
```

Sometimes, apparently randomly, bootcfg is unable to parse this:
```
api: error parsing ignition config: error at line 1, column 2
    1: --
       ^
invalid character '-' in numeric literal
```

**Comment by dghubble** [comment]
I cannot reproduce this on v0.2.0 or latest. The log message "error parsing ignition config" was not added until after v0.1.0, but removed in 68437b9e2d, before v0.2.0. Are you on a non-release, tagged image by chance? Can you update to the v0.2.0 or latest image now?

68437b9e2d was the commit which first allowed Ignition files as YAML so before that commit, you should not expect Ignition YAML to work at all. It is included in v0.2.0 if you update.

**Comment by devurandom** [comment]
Before v0.2.0 I was using a tag on the image in the Kubernetes Pod declaration. That might also explain the randomness: Probably the different nodes in my cluster had downloaded different images, some which included 68437b9 and some which didn't. I have not experienced this issue in a while, possibly since specifying the v0.2.0 tag in the Pod declaration.

You can close this issue (I don't have access to my other account today).

**Comment by dghubble** [comment]
Different versions seems like it would explain what you were seeing.
